rust/crd/src/ldap.rs

```diff
@@ -0,0 +1,241 @@
+            Some("druid_system_pass".to_string()), // TODO: replace with sed-placeholder
+        );
+        lines.insert(
+            "druid.escalator.authorizerName".to_string(),
+            Some("DruidSystemAuthorizer".to_string()),
+        );
+    }
+
+    fn add_authorizer_lines(&self, lines: &mut BTreeMap<String, Option<String>>) {
+        lines.insert(
+            "druid.auth.authorizers".to_string(),
+            Some(r#"["LdapAuthorizer", "DruidSystemAuthorizer"]"#.to_string()),
+        );
+        lines.insert(
+            "druid.auth.authorizer.DruidSystemAuthorizer.type".to_string(),
+            Some(r#"allowAll"#.to_string()),
+        );
+        lines.insert(
+            "druid.auth.authorizer.LdapAuthorizer.type".to_string(),
+            Some(r#"allowAll"#.to_string()),
+        );
+    }
+
+    pub fn generate_runtime_properties_config_lines(&self) -> BTreeMap<String, Option<String>> {
+        let mut lines: BTreeMap<String, Option<String>> = BTreeMap::new();
+
+        lines.insert(
+            "druid.auth.authenticatorChain".to_string(),
+            Some(r#"["DruidSystemAuthenticator", "Ldap"]"#.to_string()),
+        );
+
+        self.add_druid_system_authenticator_lines(&mut lines);
+        self.add_ldap_authenticator_lines(&mut lines);
+        self.add_escalator_lines(&mut lines);
+        self.add_authorizer_lines(&mut lines);
+
+        lines
+    }
+
+    fn is_ssl_enabled(&self) -> bool {
+        self.provider.tls.is_some()
+    }
+
+    fn get_ldap_protocol_and_port(&self) -> (String, u16) {
+        let protocol = if self.is_ssl_enabled() {
+            "ldaps".to_string()
+        } else {
+            "ldap".to_string()
+        };
+
+        let port = if let Some(port) = self.provider.port {
+            port
+        } else if self.is_ssl_enabled() {
+            DEFAULT_LDAP_TLS_PORT
+        } else {
+            DEFAULT_LDAP_PORT
+        };
+
+        (protocol, port)
+    }
+
+    fn credentials_validator_url(&self) -> String {
+        let (protocol, port) = self.get_ldap_protocol_and_port();
+        format!("{}://{}:{}", protocol, self.provider.hostname, port,)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use stackable_operator::commons::ldap::LdapFieldNames;
+
+    #[test]
+    fn test_ldap_settings_are_added() {
+        let ldap_settings = DruidLdapSettings {
+            provider: LdapAuthenticationProvider {
+                hostname: "openldap".to_string(),
+                port: None,
+                search_base: "ou=users,dc=example,dc=org".to_string(),
+                search_filter: "(uid=%s)".to_string(),
+                ldap_field_names: LdapFieldNames::default(),
+                bind_credentials: None,
+                tls: None,
+            },
+        };
+
+        let got = ldap_settings.generate_runtime_properties_config_lines();
+
+        assert!(got.contains_key("druid.auth.authenticator.Ldap.type"));
```

Comment:
Nit: Some more assertions would be nice but the PR is not in the state yet.
Just raising this to not forget about it

Reply:
The previous version of the code tested all fields, which seemed to err on the other side of the spectrum.

What would be useful to assert for this function in your opinion?